jupyter_server/extension/handler.py

```diff
@@ -12,19 +12,29 @@ def get_template(self, name):
         return self.settings[env].get_template(name)
 
 
-class ExtensionHandlerMixin():
-    """Base class for Jupyter server extension handlers. 
+class ExtensionHandlerMixin:
+    """Base class for Jupyter server extension handlers.
 
-    Subclasses can serve static files behind a namespaced 
-    endpoint: "/static/<extension_name>/" 
+    Subclasses can serve static files behind a namespaced
+    endpoint: "/static/<extension_name>/"
 
     This allows multiple extensions to serve static files under
-    their own namespace and avoid intercepting requests for 
-    other extensions. 
+    their own namespace and avoid intercepting requests for
+    other extensions.
     """
     def initialize(self, extension_name):
         self.extension_name = extension_name
 
+    @property
+    def extensionapp(self):
+        key = "{extension_name}".format(extension_name=self.extension_name)
+        return self.settings[key]
+
+    @property
+    def serverapp(self):
+        key = "serverapp"
+        return self.settings[key]
```

Comment:
I'm confused by the property names used here: `serverapp` is a property that maps onto `settings`. One implies a ServerApp instance, the other a settings value (not typically run-time objects?).

Reply:
Hm, the weird thing is that we pass all kinds of run-time objects to the webapp settings (kernel manager, contents manager, session manager, etc.) I agree, this probably isn't best practice but would take an extension refactor.

Reply:
So wait, what is `type(self.settings[key])` here? `ServerApp` ?